src/blob_format.cc

```
@@ -364,5 +363,79 @@ bool operator==(const BlobFileFooter& lhs, const BlobFileFooter& rhs) {
           lhs.meta_index_handle.size() == rhs.meta_index_handle.size());
 }
 
+// Seek to the specified meta block.
+// Return true if it successfully seeks to that block.
+static Status SeekToMetaBlock(InternalIterator* meta_iter,
+                              const std::string& block_name, bool* is_found,
+                              BlockHandle* block_handle = nullptr) {
+  if (block_handle != nullptr) {
+    *block_handle = BlockHandle::NullBlockHandle();
+  }
+  *is_found = true;
+  meta_iter->Seek(block_name);
+  if (meta_iter->status().ok()) {
+    if (meta_iter->Valid() && meta_iter->key() == block_name) {
+      *is_found = true;
+      if (block_handle) {
+        Slice v = meta_iter->value();
+        return block_handle->DecodeFrom(&v);
+      }
+    } else {
+      *is_found = false;
+      return Status::OK();
+    }
+  }
+  return meta_iter->status();
+}
+
+Status InitUncompressionDecoder(
+    const BlobFileFooter& footer, RandomAccessFileReader* file,
+    std::unique_ptr<UncompressionDict>* uncompression_dict,
+    std::unique_ptr<BlobDecoder>* decoder) {
+#if ZSTD_VERSION_NUMBER < 10103
+  return Status::NotSupported("the version of libztsd is too low");
+#endif
+  // 1. read meta index block
+  // 2. read dictionary
+  // 3. reset the decoder
+  assert(footer.meta_index_handle.size() > 0);
+  BlockHandle meta_index_handle = footer.meta_index_handle;
+  Slice blob;
+  CacheAllocationPtr ubuf(new char[meta_index_handle.size()]);
+  Status s = file->Read(meta_index_handle.offset(), meta_index_handle.size(),
+                        &blob, ubuf.get());
+  if (!s.ok()) {
+    return s;
+  }
+  BlockContents meta_block_content(std::move(ubuf), meta_index_handle.size());
+
+  std::unique_ptr<Block> meta(
+      new Block(std::move(meta_block_content), kDisableGlobalSequenceNumber));
+
+  std::unique_ptr<InternalIterator> meta_iter(
+      meta.get()->NewDataIterator(BytewiseComparator(), BytewiseComparator()));
+
+  bool dict_is_found = false;
```

seems it's not used. we should return an error.